### PR TITLE
Feature/improved supervisor

### PR DESCRIPTION
**Pull request by mavnn**
This still needs some more unit testing on the new supervision code, but I'm starting the pull request so you can see where things are heading.

### CODE REVIEW

File: src/Logary/Target.fs [hunk 1]
```diff
@@ -108,6 +103,19 @@ module TargetUtils =
...
         let! requests = RingBuffer.create 500u
         let shutdown = Ch ()
+        return { server   = fun _ _ -> loop metadata requests shutdown
+                 requests = requests
+                 shutdown = shutdown
+                 name     = name' }
+      }
+    }
+
+  let willAwareNamedTarget loop name : TargetConf =
```
Comment (mavnn) [inline src/Logary/Target.fs]: @haf this is where I added a new method for registering "willAware" targets so existing targets can continue to work unaltered. If you can think of a better name, let me know...

Comment (haf) [inline src/Logary/Target.fs]: @mavnn willful? ;)

**Comment by haf** [comment]
Will you have a look at @neoeinstein 's work too, @mavnn ?

**Comment by mavnn** [comment]
Yes, few nice looking tweeks I can steal

**Pull request closed without merging**